src/cluster/src/check/mod.rs

```diff
@@ -273,8 +328,53 @@ pub(crate) struct K8Version;
 
 #[async_trait]
 impl ClusterCheck for K8Version {
+    /// Check if required kubectl version is installed
     async fn perform_check(&self) -> CheckResult {
-        k8_version_check()
+        let kube_version = Command::new("kubectl")
+            .arg("version")
+            .arg("-o=json")
+            .output()
+            .map_err(CheckError::KubectlNotFoundError)?;
+        let version_text = String::from_utf8(kube_version.stdout).unwrap();
```

Comment:
You don't need to convert to string in order to convert from JSON.   JSON conversion be done from byte slice